### ISSUE TITLE
[feature] Add support for updating values of a dictionary type configuration in profile (e.g. tools.build:compiler_executables)

### ISSUE THREAD

**Issue by todorico** [conan-io/conan#13535]
### What is your suggestion?

Hi there!

I am proposing a new feature for Conan package manager that would allow updating values of a dictionary configuration like `tools.build:compiler_executables` in the same way we can already append or prepend values of a list configuration. This would make it easier to compose profiles and install compilers via `tool_requires`.

Currently, we can append or prepend values to an existing list configuration in Conan like so:

```less
# Value defined somewhere in global.conf or a main profile
tools.build:cxxflags=["--flag value"]
...
# Appending values into the existing list in some profile
tools.build:cxxflags+=["--suffix-flag suffix-value"] # -> tools.build:cxxflags=["--flag value", "--suffix-flag suffix-value"]
...
# Prepending values into the existing list in some other profile
tools.build:cxxflags=+["--prefix-flag prefix-value"] # -> tools.build:cxxflags=["--prefix-flag prefix-value", "--flag value", "--suffix-flag suffix-value"]
```

However, there is currently no way to update values of a dictionary configuration. With the proposed feature, we could update the `tools.build:compiler_executables` configuration in the same way:

```less
# Value defined somewhere in global.conf or a main profile
tools.build:compiler_executables={"c": "cl", "cpp": "cl"}
...
# Updating values by overriding into the existing dictionary in some profile
tools.build:compiler_executables**={"fortran": "ifort"} # -> tools.build:compiler_executables={"c": "cl", "cpp": "cl", "fortran": "ifort"}
...
# Updating values without overriding into the existing dictionary in some other profile (fortran is not updated because it already exists)
tools.build:compiler_executables=**{"fortran": "gfortran"} # -> tools.build:compiler_executables={"c'" "cl", "cpp": "cl", "fortran": "ifort"}
...
# Updating values without overriding into the existing dictionary in some other profile (cuda is added because it does not already exists)
tools.build:compiler_executables=**{"cuda": "nvcc"} # -> tools.build:compiler_executables={"c'" "cl", "cpp": "cl", "fortran": "ifort", "cuda": "nvcc"}
```

This feature would make it easier to compose a main profile with secondary profiles that install compilers via `tool_requires`. For example:

```less
# Main msvc profile
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.runtime=dynamic
compiler.version=192
compiler.cppstd=17
os=Windows
[conf]
tools.build:compiler_executables={'c': 'cl', 'cpp': 'cl'},
...
# Secondary ifort profile
[tool_requires]
ifort/version@company/channel
[conf]
tools.build:compiler_executables**={'fortran': 'ifort'}
...
# Secondary nvcc profile
[tool_requires]
nvcc/version@company/channel
[conf]
tools.build:compiler_executables**={'cuda': 'nvcc'}
```

Then composing the profiles would give use the desired output:

```shell
$ conan profile show -pr msvc -pr ifort -pr nvcc
Host profile:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=192
os=Windows
[tool_requires]
*: ifort/version@company/channel, nvcc/version@company/channel
[conf]
tools.build:compiler_executables={'c': 'cl', 'cpp': 'cl', 'fortran': 'ifort', 'cuda': 'nvcc'}
```

It would make my life much easier for creating usable and composable secondary compiler profiles. Whatever syntax is used for updating the dictionary values.

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
HI @todorico 

I think this might be possible, lets give it a try.

**Comment by memsharded** [comment]
It was a bit tricky, but I think that https://github.com/conan-io/conan/pull/13571 should make it (targeted for next 2.0.3 release).

Thanks for the suggestion!

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/13571 merged, it will be in next 2.0.3

**Comment by todorico** [comment]
Thank you so much ! This will be of great help to me! 😄
